content/post/practical-tb.Rmd

```diff
@@ -0,0 +1,889 @@
+  N            <- out$U+out$L+out$I+out$R  
+  rate.inc    <- 1e5*(diff(out$Incidence)/N[1:length(N)-1])
+  fr.remo     <- diff(out$Iremote)/diff(out$Incidence)
+  time         <- out$time[1:length(out$time)-1]
+  dat         <- data.frame(Years=time+(2019-400), Incidence=rate.inc)
+  dat$Sim     <- int_name
+  
+  # If it is a first run, nothing to append 
+  if (is.na(data_stub))
+  {
+    data<-dat
+  }
+  else # Append previous runs
+  {
+    data  <-rbind(data_stub, dat)
+  }
+  
+  remote<-fr.remo  # rename a variable with the fraction of remote incidence 
+  titl  <-int_name
+  
+  # Create our plot
+  p<- ggplot(data=data, mapping = aes(x=Years, y=Incidence, col=Sim))
+  p1<-p + 
+    geom_line(size=1.2) +
+    ggtitle ('TB Incidence') +
+    theme_bw() + ylab('Rate per 100,000 pop')+
+    ylim(0,max(data$Incidence))
+
+  # Pie chart of remote vs recent incidence 
+  df <- data.frame(
+    Source = c("Recent", "Remote"),
+    value  = c(1-tail(remote,1),tail(remote,1))
+  )
+  
+  mycols <- c("#0073C2FF", "#EFC000FF")
+  pie<- ggplot(df, aes(x="", y=value, fill=Source))+
+    geom_bar(width = 1, stat = "identity") +
+    coord_polar("y", start=0)+
+    scale_fill_manual(values = mycols) +
+    theme_void()+
+    ggtitle (titl) 
+  
+  output<-list("out"=out, "lines"=p1, "pie"=pie, "data"=data)
+  
+  # Spit-out results
+  return(output)
+  
+}
+
+```
+
+Set some simulation parameters 
+
+*Copy and Paste the code below*
+```{r echo=TRUE}
+
+t.intervention <- 400      # years of simulation
+t.scale        <- 3        # Scaling up time of interventions
+times  = seq(0, t.intervention, by=1)          # time scale
+
+```
+
+Create our parameter list and initial conditions
+
+*Copy and Paste the code below*
+```{r echo=TRUE}
+
+#Prepare tu run
+params <- c(break_in=break_in, selfcure=selfcure, mu=mu, mu_tb=mu_tb
+            ,beta=beta, fast=fast, imm=imm, relapse=relapse)   # running parameters
+
+# Initial Conditions
+N <- 1                  # Total population equal to unity 
+I0 <-1e-6               # TB seed at time 0
+
+xstart <- data.frame(U = N-I0,
+            L = 0,
+            I = I0,  
+            R = 0,
+            Incidence=0, 
+            Irecent=0 , 
+            Iremote=0)                
+```
+
+Now, let's solve our system 
+
+*Copy and Paste the code below*
+```{r echo=TRUE}
+
+
+#run the model
+out  <-get_intervention(xstart, params, NA,times, 
+                        NA, NA,TB.Basic, "Initial", NA) 
+#plot
+out$lines
+
+
+```
+
+  
+We have produced and incidence trajectory for a system that seems to be in endemic equilibrium. What does this mean?  
+
+
+**4) Can you take a moment and try to explain in a few words what it means for an epidemic to achieve an endemic equilibrium?**  
+
+
+**5) Making use of concepts from your previous lectures, can you think of a simple mathematical expression for the endemic equilibrium in terms of the basic reproduction number ($R_0$) ?**     
+
+<!-- --- -->
+
+<!-- ####*Answer* -->
```

Comment:
Similar to my previous comment, would you like help formatting this so that a "solutions" document can be generated? Example here:

https://github.com/reconhub/learn/blob/e39d940055f8ad6bce0f29c58d6c6f67738348ce/content/post/practical-real-time-response-2.Rmd#L166-L172